hazelcast/src/main/java/com/hazelcast/internal/serialization/InternalSerializationService.java

```diff
@@ -89,8 +89,15 @@
 
     BufferObjectDataOutput createObjectDataOutput();
 
+    /**
+     * @param data
+     * @return InternalGenericRecord if data type supports it, otherwise returns null
+     * @throws IOException
+     */
     InternalGenericRecord readAsInternalGenericRecord(Data data) throws IOException;
 
+    boolean isCompactSerializable(Object object);
```

Comment:
Consider adding javadoc for this and the method above. Ideally, I'd like javadoc on all public classes on all public methods. If it's not meant for the end user, it still helps us during maintenance or when someone is trying to navigate the codebase.